contracts/duckies/games/DuckyFamilyV1/DuckyFamilyV1_0.sol

```diff
@@ -0,0 +1,506 @@
+		Common,
+		Rare,
+		Epic,
+		Legendary
+	}
+
+	enum GeneDistributionTypes {
+		Even,
+		Uneven
+	}
+
+	// Duckling genes: Collection, Rarity, Color, Family, Body, Head, Eyes, Beak, Wings, FirstName, Temper, Skill, Habitat, Breed
+	// Zombeak genes: Collection, Rarity, Color, Family, Body, Head, Eyes, Beak, Wings
+
+	enum GenerativeGenes {
+		Collection,
+		Rarity,
+		Color,
+		Family,
+		Body,
+		Head
+	}
+
+	enum MythicGenes {
+		Collection,
+		UniqId
+	}
+
+	// ------- Internal values -------
+
+	uint8 internal constant collectionGeneIdx = 0;
+	uint8 internal constant rarityGeneIdx = 1;
+	// general genes start after Collection and Rarity
+	uint8 internal constant generativeGenesOffset = 2;
+
+	// number of values for each gene for Duckling and Zombeak collections
+	uint8[][2] internal collectionsGeneValuesNum;
+	// distribution type of each gene for Duckling and Zombeak collections
+	uint32[2] internal collectionsGeneDistributionTypes;
+
+	uint8 internal mythicAmount;
+
+	// chance of a Duckling of a certain rarity to be generated
+	uint8[] internal rarityChances; // 70, 20, 5, 1
+
+	// chance of a Duckling of certain rarity to mutate to Zombeak while melding
+	uint8[] internal collectionMutationChances; // 10, 5, 2, 0
+
+	uint8[] internal geneMutationChance; // 955, 45 (4.5% to mutate gene value)
+	uint8[] internal geneInheritanceChanges; // 5, 4, 3, 2, 1
+
+	// ------- Public values -------
+
+	uint8 public constant MAX_PACK_SIZE = 100;
+	uint8 public constant FLOCK_SIZE = 5;
+
+	ERC20BurnableUpgradeable public duckiesContract;
+	IDucklings public ducklingsContract;
+	address public treasureVaultAddress;
+
+	uint256 public mintPrice;
+	uint256 public meldPrice;
+
+	CountersUpgradeable.Counter public nextMythicId;
+	CountersUpgradeable.Counter public nextMythicZombeakId;
+
+	// ------- Initializer -------
+
+	function initialize(
+		address duckiesAddress,
+		address ducklingsAddress,
+		address treasureVaultAddress_
+	) external initializer {
+		__AccessControl_init();
+		__UUPSUpgradeable_init();
+		__Random_init();
+
+		_grantRole(DEFAULT_ADMIN_ROLE, msg.sender);
+		_grantRole(UPGRADER_ROLE, msg.sender);
+		_grantRole(MAINTAINER_ROLE, msg.sender);
+
+		duckiesContract = ERC20BurnableUpgradeable(duckiesAddress);
+		ducklingsContract = IDucklings(ducklingsAddress);
+		treasureVaultAddress = treasureVaultAddress_;
+
+		mintPrice = 5 * 10 ** duckiesContract.decimals();
+		meldPrice = 5 * 10 ** duckiesContract.decimals();
+
+		// config
+		// duckling
+		// TODO: confirm gene values
+		collectionsGeneValuesNum[0] = [4, 5, 9, 28, 30, 14, 10, 36, 16, 12, 5, 28];
+		collectionsGeneDistributionTypes[0] = 2940; // reverse(001111101101) = 101101111100
+		// zombeak
+		// TODO: confirm gene values
+		collectionsGeneValuesNum[1] = [2, 3, 7, 6, 9, 7, 10, 4, 4, 4, 4, 4];
+		collectionsGeneDistributionTypes[1] = 2940; // reverse(001111101101) = 101101111100
+
+		// TODO: confirm
+		mythicAmount = 65;
+
+		rarityChances = [70, 20, 5, 1];
```

Comment:
This is incorrect, the total must be 100,
chance of legendary is 1/100 not 1/94